public/app/core/components/AppChrome/DockedMegaMenu/MegaMenu.tsx

```diff
@@ -48,8 +56,27 @@ export const MegaMenu = React.memo(
         <nav className={styles.content}>
           <CustomScrollbar showScrollIndicators hideHorizontalTrack>
             <ul className={styles.itemList}>
-              {navItems.map((link) => (
-                <MegaMenuItem link={link} onClose={onClose} activeItem={activeItem} key={link.text} />
+              {navItems.map((link, index) => (
+                <Flex key={link.text} direction="row" alignItems="center">
+                  <MegaMenuItem
+                    link={link}
+                    onClick={state.megaMenu === 'open' ? onClose : undefined}
+                    activeItem={activeItem}
+                  />
+                  {index === 0 && (
```

Comment:
yep gonna add this as an e2e test at the end since i wanna check the docking persists between page transitions/page reloads etc 👍 have added it to the epic here: https://github.com/grafana/grafana/issues/72446